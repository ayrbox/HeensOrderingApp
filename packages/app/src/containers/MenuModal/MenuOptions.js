import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Checkbox from '@material-ui/core/Checkbox';

const styles = {
  checkBox: {
    padding: 0,
  },
  optionCost: {
    width: '200px',
    textAlign: 'right',
  },
};

const MenuOptions = ({
  classes,
  options,
  onToggleSelect,
}) => (
  <List data-testid="menu-option-list">
    {options.map(({
      id,
      description,
      additionalCost,
      isChecked,
    }) => (
      <ListItem
        key={id}
        role="button"
        dense
        button
        onClick={onToggleSelect(id)}
        data-testid="menu-option-item"
      >
        <Checkbox
          tabIndex={-1}
          disableRipple
          className={classes.checkBox}
          checked={isChecked}
        />
        <ListItemText primary={description} />
        <ListItemText
          primary={additionalCost === 0 ? '' : `Â£${additionalCost.toFixed(2)}`}
          className={classes.optionCost}
        />
      </ListItem>
    ))}
  </List>
);

MenuOptions.propTypes = {
  options: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  classes: PropTypes.shape().isRequired,
  onToggleSelect: PropTypes.func.isRequired,
};

export default withStyles(styles)(MenuOptions);
