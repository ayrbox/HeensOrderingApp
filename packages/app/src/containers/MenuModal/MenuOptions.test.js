import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import MenuOptions from './MenuOptions';

const sampleOptions = [{
  additionalCost: 1,
  id: '5d65b14b0090bc34b87133ff',
  description: 'Chilli Sauce',
}, {
  additionalCost: 0.5,
  id: '5d65b14b0090bc34b87133fe',
  description: 'Tomato Ketchup',
}];

const mockToggle = jest.fn();

describe('<MenuOptions />', () => {
  describe('when options given', () => {
    it('should render correct', () => {
      const {
        container,
        getByTestId,
        getAllByTestId,
      } = render(
        <MenuOptions
          options={sampleOptions}
          onToggleSelect={mockToggle}
        />,
      );

      getByTestId('menu-option-list');

      const items = getAllByTestId('menu-option-item');
      expect(items.length).toEqual(sampleOptions.length);

      items.forEach((item) => {
        fireEvent.click(item);
      });

      expect(mockToggle).toHaveBeenCalledTimes(sampleOptions.length);
      expect(container).toMatchSnapshot();
    });
  });
});
