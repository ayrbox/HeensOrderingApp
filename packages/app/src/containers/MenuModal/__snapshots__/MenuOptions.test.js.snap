// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuOptions /> when options given should render correct 1`] = `
<div>
  <ul
    class="MuiList-root-3 MuiList-padding-4"
    data-testid="menu-option-list"
  >
    <div
      class="MuiButtonBase-root-19 MuiListItem-root-7 MuiListItem-default-10 MuiListItem-dense-11 MuiListItem-gutters-15 MuiListItem-button-16"
      data-testid="menu-option-item"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiButtonBase-root-19 MuiIconButton-root-32 MuiPrivateSwitchBase-root-28 MuiCheckbox-root-22 MuiCheckbox-colorSecondary-27 MenuOptions-checkBox-1"
      >
        <span
          class="MuiIconButton-label-37"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-38"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
          <input
            class="MuiPrivateSwitchBase-input-31"
            data-indeterminate="false"
            tabindex="-1"
            type="checkbox"
            value=""
          />
        </span>
      </span>
      <div
        class="MuiListItemText-root-47 MuiListItemText-dense-49"
      >
        <span
          class="MuiTypography-root-53 MuiTypography-subheading-60 MuiListItemText-primary-50 MuiListItemText-textDense-52"
        >
          Chilli Sauce
        </span>
      </div>
      <div
        class="MuiListItemText-root-47 MuiListItemText-dense-49 MenuOptions-optionCost-2"
      >
        <span
          class="MuiTypography-root-53 MuiTypography-subheading-60 MuiListItemText-primary-50 MuiListItemText-textDense-52"
        >
          £1.00
        </span>
      </div>
      <span
        class="MuiTouchRipple-root-89"
      />
    </div>
    <div
      class="MuiButtonBase-root-19 MuiListItem-root-7 MuiListItem-default-10 MuiListItem-dense-11 MuiListItem-gutters-15 MuiListItem-button-16"
      data-testid="menu-option-item"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiButtonBase-root-19 MuiIconButton-root-32 MuiPrivateSwitchBase-root-28 MuiCheckbox-root-22 MuiCheckbox-colorSecondary-27 MenuOptions-checkBox-1"
      >
        <span
          class="MuiIconButton-label-37"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-38"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
          <input
            class="MuiPrivateSwitchBase-input-31"
            data-indeterminate="false"
            tabindex="-1"
            type="checkbox"
            value=""
          />
        </span>
      </span>
      <div
        class="MuiListItemText-root-47 MuiListItemText-dense-49"
      >
        <span
          class="MuiTypography-root-53 MuiTypography-subheading-60 MuiListItemText-primary-50 MuiListItemText-textDense-52"
        >
          Tomato Ketchup
        </span>
      </div>
      <div
        class="MuiListItemText-root-47 MuiListItemText-dense-49 MenuOptions-optionCost-2"
      >
        <span
          class="MuiTypography-root-53 MuiTypography-subheading-60 MuiListItemText-primary-50 MuiListItemText-textDense-52"
        >
          £0.50
        </span>
      </div>
      <span
        class="MuiTouchRipple-root-89"
      />
    </div>
  </ul>
</div>
`;
